[Papattarada A.] iP

Punpun1643 opened the pull request:
# DukeyDuke
> DukeyDuke is a personal todo list assistant which is Command Line Interface (CLI) optimized

The source code could be accessed [here](https://github.com/Punpun1643/ip.git)

The following are the features that I have implemented: 
- Add and Delete a task
- Mark task as done
- Unmark task as not done
- Add *todo, event, deadline* as task types
- Modify date and time format

The following are the steps to run the Duke jar file (the jar file could be found on the most recent release):
1. Copy the jar file into an empty folder
2. Open a command window in that folder
3. Run the command `java -jar {filename}.jar` (i.e. run the command in the same folder as the jar file)

The `main` method can be located at the Duke.java file. Here's the `main` method:
```java
public class Duke {
    public static void main(String[] args) throws IOException, DukeException {
        ...
    }
}
``` 

Upcoming features for the **next release** :smiley:: 
- [x] Use Gradle
- [x] Use Checkstyle
- [ ] GUI
- [ ] \(Optional) fix bugs

Commit:
Add Gradle support

Punpun1643 committed:
Skeletal version

Punpun1643 committed:
Add divider and indentation

Punpun1643 committed:
Level 1

Punpun1643 committed:
Level-2. Add, List

Punpun1643 committed:
Add divider to list

Punpun1643 committed:
Add Task.java

Punpun1643 committed:
Fix count increment bug

Punpun1643 committed:
Add mark as undone method

Punpun1643 committed:
Mark as Done and Undone

Punpun1643 committed:
Add the use of inheritance to support multiple task types

Punpun1643 committed:
Remove getStatus method

Punpun1643 committed:
Level 4. ToDos, Events, Deadlines. Add support for tracking these types of tasks

Punpun1643 committed:
Edit logic for input tasks that may start with the word mark

Punpun1643 committed:
Remove space from unmark

Punpun1643 committed:
Automated Text UI Testing

Punpun1643 committed:
A-Exceptions. Use exceptions to handle errors

Punpun1643 committed:
Level 5. Handle Errors

Punpun1643 committed:
A-Collections. Use java collections classes to store task.

Punpun1643 committed:
Update task validator to handle new exceptions for delete command

Punpun1643 committed:
Level-6. Delete

Punpun1643 committed:
A-Enums. Create enums for TaskCommand

Punpun1643 committed:
Modify Deadline.java to support saving to duke.txt

Punpun1643 committed:
Modify Event.java to support saving to duke.txt

Punpun1643 committed:
Modify to support saving to duke.txt

Punpun1643 committed:
Modify to support saving to duke.txt

Punpun1643 committed:
Add DukeTaskSaver.java to save task to duke.txt

Punpun1643 committed:
Modify to support saving of todoList tasks to duke.txt

Punpun1643 committed:
Modify to support dates and times

Punpun1643 committed:
Modify to support loading

Punpun1643 committed:
Resolve conflict import

Punpun1643 committed:
Resolve conflict

Punpun1643 committed:
Implement OOP for Storage

Punpun1643 committed:
Implement OOP for Parser

Punpun1643 committed:
Implement TaskList and Ui OOP

Punpun1643 committed:
Put all classes in one pacjage duke

Punpun1643 committed:
Add Junit Tests

Punpun1643 committed:
Create a JAR file

Punpun1643 committed:
Add find feature

Punpun1643 committed:
Implement find

Punpun1643 committed:
Comply to coding standard

Punpun1643 committed:
Add JavaDoc comments

Punpun1643 committed:
Follow coding standard

Punpun1643 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

Punpun1643 committed:
Set up gradble to run unit tests

Punpun1643 committed:
Use checkstyle

shurvirarora commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,138 @@
+
+            // create new file
+            if (f.createNewFile()) {
+                System.out.println(this.relativePath + " file created in the project root directory");
```
Perhaps length of line could be shorter (e.g line break before operator) to abide by Java Coding Standard 👍

raynerljm replied:
Yup I agree, would be good to break and let the '+' be at the front of the new line

shurvirarora commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,125 @@
+        System.out.println("        " + taskLists.get(num - 1).toString());
+        taskLists.remove(num - 1);
+
+        String taskRemainingString = String.format("    Now you have %d tasks in the list.", taskLists.size());
```
A line break could be made here after the comma 👍

Shivang211 replied:
Agreed, might make the code more readable

shurvirarora commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,56 @@
+import java.time.LocalTime;
+import java.time.Month;
+
+public class Deadline extends Task {
```
You've avoided a common mistake by making sure there is a white spacing between the class name "Task" and '{' as many overlooked this. 👍

shurvirarora reviewed (commented):
Overall, Java Coding Standard is followed and the code looks good to me! Slight nits to fix regarding with line breaks for long lines.

raynerljm commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,56 @@
+package duke;
+
+import java.time.DayOfWeek;
```
Good job not using wildcard imports

raynerljm commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,56 @@
+    protected LocalTime time;
+
+    /**
+     * Constructs a Deadline object.
```
Good job doing JavaDocs and putting a fullstop after it

raynerljm commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,228 @@
+     */
+    public static void main(String[] args) throws IOException, DukeException {
+        // divider
+        final String LINES = "    ---------------------------------";
```
Good job using screaming snake case for constants

raynerljm commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,228 @@
+        Scanner sc = new Scanner(System.in);
+
+        // initializing variable
+        boolean isBye = false;
```
Good job using 'is' for boolean variables

raynerljm commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,228 @@
+        boolean isBye = false;
+
+        // array container
+        ArrayList<Task> todoLists = new ArrayList<Task>();
```
Good job using a plural variable name for collections

raynerljm commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,228 @@
+                }
+
+                switch (userCommand) {
+                case "list":
```
Good job with the switch case indentation levels

raynerljm commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,55 @@
+     * @return
+     */
+    public String toStringForSave() {
+        return (isDone ? "# 1 # " : "# 0 # ") + this.description;
```
Note you do not need to put 'this' before description as there is no ambiguity

raynerljm reviewed (approved):
Really good stuff! LGTM

Punpun1643 committed:
Implement GUI

Punpun1643 committed:
Use varargs

Shivang211 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,382 @@
+import java.util.Arrays;
+
+public class Parser {
+    private static final ArrayList<String> VALID_USER_COMMAND = new ArrayList<String>(Arrays.asList("todo", "event",
```
Good job to avoid lines greater than 100 chars

Shivang211 reviewed (commented):
A great job overall, keep up the good work!

chanyijuan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,29 @@
+        while (!isBye) {
+            if (userInput.equals("bye")) {
+                isBye = true;
+                System.out.println("Bye. See you again next time! Have a nice day 😊!");
```
Would be great if you could do what you did earlier with the greeting String and line-break and + here!

chanyijuan commented on src/main/java/Duke.java:
```diff
@@ -15,9 +15,22 @@ public static void main(String[] args) {
+        /**
+         * current number of to do tasks
+         */
+        int count = 0;
```
Maybe instead of describing count variable in the javadoc comments you can name it something like "numOfTasks"?

chanyijuan commented on src/main/java/duke/Duke.java:
```diff
@@ -30,28 +30,27 @@ public static void main(String[] args) throws IOException, DukeException {
         try {
-            todoList = storage.load();
-            taskList = new TaskList(todoList);
+            todoLists = storage.load();
+            taskLists = new TaskList(todoLists);
```
Great effort changing even variable names to plural to comply with the naming standard!

chanyijuan commented on src/main/java/Duke.java:
```diff
@@ -10,16 +10,12 @@
-
         // divider
-        String lines = "    ---------------------------------";
+        String LINES = "    ---------------------------------";
```
Nice change to comply with static final variables 😊

chanyijuan commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,79 @@
+        String taskString = String.format("%s", taskList.get(taskToUnmark - 1).toString());
+        System.out.println("    " + taskString);
+        System.out.println(LINES);
+    }
```
Just a personal preference, but maybe the parameter can be renamed to just "task". That's just my taste though!

Punpun1643 committed:
Fix UI

Punpun1643 committed:
Update build.gradle

The build of JAR file is done using Gradle.
Make necessary change to the build.gradle configuration to
support this.

Modify the mainClassName.

The application is currently launch by the launcher due to
the implementation of the GUI.

Punpun1643 committed:
Use assertions

Initially no assertions is used.

Add assertions to necessary methods to make sure that
they work as per expected and to verify assumptions
and to detect any possible programming errors.

Let's
* use assertions explicitly to verify assumptions.

Punpun1643 committed:
Modify assertion

Punpun1643 committed:
Improve code quality

Punpun1643 committed:
Merge pull request #1 from Punpun1643/branch-A-Assertions

Use assertions

Punpun1643 committed:
Merge branch 'master' into branch-A-CodeQuality

Punpun1643 committed:
Merge pull request #2 from Punpun1643/branch-A-CodeQuality

Improve code quality

Punpun1643 committed:
Comply to coding standard

Punpun1643 committed:
Implement tag for todo task

Punpun1643 committed:
Merge pull request #3 from Punpun1643/branch-BCD-Extension

Implement tag for todo task

Punpun1643 committed:
Implement C-Tagging

Punpun1643 committed:
Set up CI

Punpun1643 committed:
Fix format

Punpun1643 committed:
Remove CI

Punpun1643 committed:
Better GUI

Punpun1643 committed:
Finalise checkstyle

Punpun1643 committed:
Add Ui.png

Punpun1643 committed:
Add documentation

Punpun1643 committed:
Release v0.2

Punpun1643 committed:
Update dependencies

Punpun1643 committed:
Fix bug for bye command

Punpun1643 committed:
Fix method lengths

Punpun1643 committed:
Fix method size for the Storage class

Punpun1643 committed:
Merge pull request #5 from Punpun1643/fix-method-length

Fix method length

Punpun1643 committed:
Resolve long methods

Punpun1643 committed:
Fix method length in Duke

Punpun1643 committed:
Fix README

Punpun1643 committed:
Fix README bug

Punpun1643 committed:
Merge pull request #7 from Punpun1643/fix-method-length

Resolve long methods

Punpun1643 committed:
Add assertion

Punpun1643 committed:
Fix minor bug on description
